app/Http/Controllers/CommentController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use App\Post;
+use App\Comment;
+
+class CommentController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request, $postId)
+    {
+        $request->validate([
+            'body' => 'required|min:2|max:2000',
+        ]);
+
+        $user = Auth::user();
+        $post = Post::findOrFail($postId);
+
+        $comment = new Comment();
+        $comment->body = $request->body;
+        $comment->parent_id = $request->get('parent_id', null);
+        $comment->post_id = $post->id;
+        $comment->user_id = $user->id;
+        $comment->post()->associate($post);
```

Comment:
You need `$comment->post_id = $post->id;` or `$comment->post()->associate($post);` - they do same work

Reply:
So you want to say if I delete this line of code:
$comment->post()->associate($post);
nothing changes, right?